clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1235,19 +1245,6 @@ private void close(Duration timeout, boolean swallowException) {
         log.debug("Kafka producer has been closed");
     }
 
-    private static Map<String, Object> propsToMap(Properties properties) {
```

Comment:
this check is duplicate to https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java#L103